Flake: tuple iterator tests

```
panic: Tuple iterator garbage collected before Close() was called

goroutine 5 [running]:
github.com/authzed/spicedb/internal/datastore/common.MustIteratorBeClosed(0xc00045c640)
	/home/runner/work/spicedb/spicedb/internal/datastore/common/tuple.go:75 +0x65
FAIL	github.com/authzed/spicedb/internal/datastore/proxy	11.180s
```

--- a/internal/datastore/proxy/hedging_test.go
+++ b/internal/datastore/proxy/hedging_test.go
@@ -325,6 +325,7 @@ func TestDatastoreE2E(t *testing.T) {
 		},
 	)
 	require.NoError(err)
+	defer it.Close()
 
 	only := it.Next()
 	require.Equal(expectedTuples[0], only)

